aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -95,6 +110,70 @@ struct FusedOptimizerTensorListMetadata {
   int start_tensor_this_launch;
 };
 
+bool can_use_static_tensor_list_meta(
+    std::vector<std::vector<at::Tensor>>& tensor_lists,
+    int depth) {
+  const int64_t n_tensors = tensor_lists[0].size();
+  if (n_tensors > depth_to_max_tensors[depth - 1]) {
+    return false;
+  }
+  int64_t num_blocks = 0;
+  for (const auto t : c10::irange(n_tensors)) {
+    const auto numel = tensor_lists[0][t].numel();
+    const auto chunks = numel / kChunkSize + (numel % kChunkSize != 0);
```

Comment:
https://stackoverflow.com/questions/2745074/fast-ceiling-of-an-integer-division-in-c-c

After reading alot of this, I disagree with the suggestion lol

Reply:
godbolt doesn't seem to agree though: https://godbolt.org/z/1TMfEhM67

Reply:
At this point in the code, have the 0 numel tensors been filtered? If I am not mistaken, it seems like not. In that case, @albanD 's approach has `chunks = 1`, whereas @yifuwang's approach has `chunks = 0` (https://godbolt.org/z/o8a5vfqGh).

Reply:
Andrew points out a good semantic difference but in terms of perf on positive ints
https://quick-bench.com/q/G7tjb5ZDp1Z6MXn0WVYgy4Dz3Vk
I stand mistaken and admit defeat

Anyways we should have a ceil_div function in c10

Reply:
Wait @drisspg isn't the second bar slowest in your quick-bench?

Reply:
It is worth checking the non-divisible case too: https://quick-bench.com/q/ZveVJxeF6oBfmrux8T-Ajadn6rQ

Reply:
OHHHHH shig  you are right @janeyx99... never mind I claim victory on all my foes!!!

Reply:
LOL @awgu  throwing in a wrench --> the nondivisible case is unfortunately higher I would guess.

Reply:
I think that one run was an anomaly. I re-ran it a few times, and the time came back down to normal. I cannot seem to reproduce the much slower `div_ceil1`.

Either way, `div_ceil1` handles dividing 0 in the correct way, so maybe we should just keep it as is in the PR.

Reply:
https://quick-bench.com/q/LAg-v-n4HlJ2DfrQsfDdZROV-HE

Reply:
So it turns out that we do have a ceil_div in ATen: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/ceil_div.h

The question now, is this the correct implementation?

Reply:
Hhahahaha that's what I think I started with before @r-barnes told me it had risk of overflow and so I corrected it to be the current state it is.

Reply:
btw I do think a lot of the benchmark shared here were not super meaningful as all the variables were optimized away.
https://quick-bench.com/q/3ldZe15ZJerQ6oYrmw9AcctBM5w should have something more meaningful but it is very very noisy still.

I would say that it seems to agree with the implementation in ceil_div.h is usually faster.
Also from checking the assembly, it is finding out by itself that it can do a bitshift instead of a full division when the second argument is constant (which it is in our case).

Reply:
Switched to `at::ceil_div` for now. We can move the battlefield there :)